rest/admin_api_test.go

```diff
@@ -1060,7 +1060,7 @@ func TestReplicate(t *testing.T) {
 	var rt restTester
 
 	//Initiate synchronous one shot replication
-	assertStatus(t, rt.sendAdminRequest("POST", "/_replicate", `{"source":"http://localhost:4985/db", "target":"http://localhost:4985/db"}`), 500)
+	assertStatus(t, rt.sendAdminRequest("POST", "/_replicate", `{"source":"http://localhost:4985/db", "target":"http://localhost:4985/db"}`), 200)
 
 	//Initiate asyncronous one shot replication
 	assertStatus(t, rt.sendAdminRequest("POST", "/_replicate", `{"source":"http://localhost:4985/db", "target":"http://localhost:4985/db", "async":true}`), 200)
```

Comment:
I feel like we need a unit test that would reproduce the original bug (#1656), to ensure this doesn't get broken again.